reckoner/chart.py

```diff
@@ -191,63 +185,168 @@ def install(self, namespace=None, context=None):
+
+        # Add the version arguments
         if self.version:
             self.args.append('--version={}'.format(self.version))
-        for file in self.files:
-            self.args.append("-f={}".format(file))
-
-        for key, value in self.values.iteritems():
-            for k, v in self._format_set(key, value):
-                self.args.append("--set={}={}".format(k, v))
 
+        # HACK: This is in place until we can fully deprecate the usage of set
+        #       in place of values. Currently values: gets translated into
+        #       --set commands to the helm command line run. This should be
+        #       changed so that set: goes to command line args and values: goes
+        #       to temporary values files, to keep strong types in yaml.
+        #       This will go away once we fully change the functionality of
+        #       values: vs sets: settings in course.yml
+        self._merge_set_and_values()
+
+        # Build the list of existing yaml files to use for the helm command
+        self.build_files_list()
+
+        # Build the file arguments from the `values: {}` in course.yml
+        self.build_temp_values_files()
+
+        # Build the list of --set arguments
+        self.build_set_arguments()
+
+        # Build the list of --set-string arguments
+        self.build_set_string_arguments()
+
+    def _merge_set_and_values(self):
+        '''
+        This is a temporary method that will be gone once the values: vs set:
+        debacle has been resolved. (See https://github.com/reactiveops/reckoner/issues/7)
+
+        NOTE ONLY RUN THIS ONCE BECAUSE IT'S NOT IDEMPOTENT
+        '''
+        if self._hack_set_values_already_merged:
+            raise StandardError('This method cannot be called twice. '
+                                'If you are seeing this please open an '
+                                'issue in github.')
+
+        def merge_dicts(values, sets):
+            '''This does a dict merge and prefers "sets" values'''
+            new_dict = values.copy()
+            new_dict.update(sets)
+            return new_dict
+
+        logging.debug('Merging values: into sets: - sets: take precedence.')
+        logging.debug('Original value of sets: {}'.format(self.set_values))
+        self.set_values = merge_dicts(self.values, self.set_values)
+        logging.debug('New value of sets: {}'.format(self.set_values))
+        self._hack_set_values_already_merged = True
+
+    def build_temp_values_files(self):
+        '''
+        This method builds temporary files based on the values: settings
+        provided in the course.yml. This effectively keeps type persistence
+        between the course.yml and what is passed to helm. If you use set:
+        value arguments then you can lose types like int, float and true/false
+        '''
+        if self.values:
+            self._deprecation_messages = [
+                "DEPRECATION NOTICE: The behavior of `values: {}` will change in "
```

Comment:
I'm not certain we need quite this much info in the message.  As a user, I find overly long deprecation messages annoying and tend to skip over them.  The link to the issue is really good and will give most of this information to the user.

Reply:
Ok - I can summarize this a bit by using `"DEPRECATION NOTICE: Change `values: {}` to `set-values: {}` to keep consistent behavior beyond v1.x. Details : github.com/....."`

Sound ok?

Reply:
That sounds awesome.